langserver/handler.go

```diff
@@ -392,7 +397,7 @@ func (h *LangHandler) Handle(ctx context.Context, conn jsonrpc2.JSONRPC2, req *j
 				// a user is viewing this path, hint to add it to the cache
 				// (unless we're primarily using binary package cache .a
 				// files).
-				if !h.Config.UseBinaryPkgCache {
+				if !h.Config.UseBinaryPkgCache || req.Method == "textDocument/didSave" {
```

Comment:
the main issue with enabling typechecking again, is that it uses a lot of memory. This might be a blocker for this PR / reason to make this opt-in. WDYT @slimsag

Reply:
I have added a simple command line flag `--diagnostics` to activate it